scripts/suite_objects.py

```diff
@@ -1200,6 +1205,12 @@ def analyze(self, phase, group, scheme_library, suite_vars, level):
                                                   vstdname))
                 # end if
             # end if
+            # Are there any forward/reverse transforms for this variable?
+            if compat_obj is not None and (compat_obj.has_vert_transforms or
+                                           compat_obj.has_unit_transforms or
+                                           compat_obj.has_kind_transforms):
+                self.add_var_transform(var, compat_obj, vert_dim)
```

Comment:
do we need to do something if the variables aren't compatible at all? I see that we're logging the reason for incompatibility in compatible() but I don't see an endrun or anything like that.

Reply:
It happens much earlier, during __init in VarCompatObj, if there are no valid conversions it will report errors then. For example,
-- Unsupported unit conversion, 'umsdsxc' to 'm' for 'effective_radius_of_stratiform_cloud_ice_particle'
-- Unknown DDT type, real_crazy, at /glade/u/home/dswales/Projects/framework/capgen/unit_conversions/ccpp-framework/test/var_action_test/effr_calc.meta:42

Reply:
Ah yes, thanks Dustin!

Follow-up question, how does it work for what the code calls "fatal incompatibilities" like standard name and type? It looks like, for those, the incompatibility reasons are logged but it the logic (in the init phase of VarCompatObj) precludes an error from being thrown (errors are thrown by _get_kind_convstrs).

Reply:
The fatal incompatibilities are caught before we try to do a transform, if "not found".

Reply:
ok. Thanks again, Dustin!